src/bt/composites/MemSelector.cpp

```diff
@@ -9,29 +9,33 @@
 
 namespace bt {
 
-void MemSelector::initialize() {
-    index = 0;
-}
+    MemSelector::MemSelector(nvector children) : Composite(children) {}
 
-Node::Status MemSelector::update() {
-    if (HasNoChildren()) {
-        return Status::Success;
+    MemSelector::MemSelector() {}
+
+    void MemSelector::initialize() {
+        index = 0;
     }
 
-    // Keep going until a child behavior says it's running.
-    while (index < children.size()) {
-        auto &child = children.at(index);
-        auto status = child->tick(world, field);
+    Node::Status MemSelector::update() {
+        if (HasNoChildren()) {
+            return Status::Success;
+        }
+
+        // Keep going until a child behavior says it's running.
+        while (index < children.size()) {
+            auto &child = children.at(index);
```

Comment:
dont use .at(), it has overhead + can throw an exception, you already check whether index < children.size() anyway so just use operator[]

Reply:
Will fix